corehq/apps/userreports/views.py

```diff
@@ -1125,6 +1126,14 @@ def post(self, request, *args, **kwargs):
                 )
         return self.get(request, *args, **kwargs)
 
+    def get_reports(self):
+        if self.read_only:
```

Comment:
It's possible for nonstatic reports to reference a static data source, and technically possible (although rare) for a static report to refernece a nonstatic data source.  We probably want to check nonstatic and static report configurations regardless of where the data source is static or not.